docs/testing.md

```diff
@@ -0,0 +1,72 @@
+# OpenShift on Azure testing
+
+OpenShift on Azure project is using few testing methods. 
+Some of the test code base is being reused from other projects.
+
+PR commands to execute tests:
+```
+/test e2e - run native e2e tests
+/test unit - run unit tests
+/test build - run container build tests
+/test upgrade - run upgrade cluster tests
+/test conformance - run origin conformance tests
+```
+
+## Unit test
+
+Packages under `pkg` directory contains their individual unit tests.
+To run all unit tests locally, execute: `make unit`.
+To run individual test:
+`go test github.com/openshift/openshift-azure/pkg/tls`
```

Comment:
To run a single package's tests:
`go test ./pkg/tls`

To run a subset of tests in a package:
`go test -run TestFoo ./pkg/tls`